src/Input.jsx

```diff
@@ -0,0 +1,67 @@
+import React, { PropTypes } from 'react';
+import classNames from 'classnames';
+
+const propTypes = {
+  children: PropTypes.node,
+  type: PropTypes.string,
+  size: PropTypes.string,
+  state: PropTypes.string,
+  tag: PropTypes.string,
+  'static': PropTypes.bool,
+  className: PropTypes.string,
+};
+
+const defaultProps = {
+  tag: 'p',
+  type: 'text',
+};
+
+const Form = (props) => {
+  const {
+    className,
+    type,
+    size,
+    state,
+    tag,
+    'static': staticInput,
+    ...attributes,
+  } = props;
+
+  const checkInput = ['radio', 'checkbox'].indexOf(type) > -1;
+
+  const fileInput = type === 'file';
+  const textareaInput = type === 'textarea';
+  const selectInput = type === 'select';
```

Comment:
Should `textarea` and `select` get moved into their own components? 

- I think it's super convenient wrapping this all in one component,
- but I also think it might be misleading to people since that's not a valid type attribute (yeah we get flexibility with react components but it's something to think about)